okta-idx-kotlin/src/main/java/com/okta/idx/kotlin/client/InteractionCodeFlow.kt

```diff
@@ -50,19 +51,20 @@ class InteractionCodeFlow internal constructor(
         /**
          * Used to create an InteractionCodeFlow, and to start an authorization flow.
          */
-        suspend fun OidcClient.createInteractionCodeFlow(
+        suspend fun create(
```

Comment:
InteractionCodeFlow instantiation is a bit different from other flows because it performs some network calls on initialization (so that's why it's a suspend create instead of constructor invocation). Let me know if this should be changed to a constructor based invocation instead.

Reply:
Also, I see that Swift SDK has a constructor, and a start function. We could do something similar here

Reply:
Yes, in general the authentication flows should have a similar pattern whereby flows have a:

* Initializer
* `start` function which kicks off the flow, and
* `resume` function that continues the flow for subsequent requests.

https://okta.github.io/okta-mobile-swift/development/oktaoauth2/documentation/oktaoauth2/introductiontoauthenticationflows

InteractionCodeFlow, as you say, is a bit different because it is so highly introspective, but I think separating the network setup work into a `start` function is a good practice, so the initializer doesn't have to be async.  But we can address that later.

Reply:
One more thing: since this is most likely going to need to be a major version number bump (owing to the breaking API changes caused by the underlying type changes), would it make sense to align InteractionCodeFlow to the patterns of other authentication flows?  e.g. how difficult would it be to move the async operations to a separate `start` function, allowing the client and its required configuration to be supplied to the initializer?

Also, perhaps I haven't noticed this before (for example, on the AuthorizationCodeFlow implementation), but the redirectUrl is a plain string there too. Is there a URL datatype that can be used, which can help to ensure that the value supplied is a properly-formed URN/URI?

Reply:
Changing InteractionCodeFlow to use a start function makes sense and wouldn't be hard to do at all.

There is a URL datatype that we can use: [HttpUrl](https://square.github.io/okhttp/3.x/okhttp/okhttp3/HttpUrl.html). I can create two versions of start function and mark the string URL version as deprecated. We can do something similar in okta-mobile-kotlin.

Reply:
That's great! 😄  Before you do, would you mind taking a look at the draft Client SDK Design site I created, and see the section on authentication flows?